python/ray/tests/aws/test_autoscaler_aws.py

```diff
@@ -225,17 +226,38 @@ def test_fills_out_amis(iam_client_stub, ec2_client_stub):
     ec2_client_stub.assert_no_pending_responses()
 
 
-def test_temp_create_sg_multinode(iam_client_stub, ec2_client_stub):
+def test_missing_key(iam_client_stub):
+    """
+    Check that an error is raised when ssh_private_key is given in the
+    auth_config but a node type is missing KeyName.
+    """
+    config = helpers.load_aws_example_config_file("example-full.yaml")
+    config["auth"]["ssh_private_key"] = "~/fake/path"
+    config["available_node_types"]["ray.head.default"]["node_config"][
+        "KeyName"] = "fakeKey"
+
```

Comment:
Lets explicitly unset the "ray.worker_default" to make it easier to read this code and understand that there is another key that does not have a KeyName.